vmdb/lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -88,9 +88,15 @@ def self.expose(*args)
         next if method_name.to_sym == :id
         self.association = method_name if options[:association]
         define_method(method_name) do |*params|
-          ret = object_send(options[:method] || method_name, *params)
+          method = options[:method] || method_name
+          ret = object_send(method, *params)
           return options[:override_return] if options.has_key?(:override_return)
-          return wrap_results(ret)
+          reflection = @object.class.reflection_with_virtual(method)
+          if reflection && reflection.collection?
```

Comment:
I think this check should be inside wrap_results.  wrap_results is a general "just wrap this thing" method, and inside it decides which way to go.